akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -295,6 +294,11 @@ abstract class RouteDirectives extends RespondWithDirectives {
     D.complete(value.asScala.fast.map(v => ToResponseMarshallable(v)(marshaller)).recover(unwrapCompletionException))
   }
 
+  def fromFunction(handler: akka.japi.function.Function[HttpRequest, CompletionStage[HttpResponse]]): Route = {
+    import akka.http.impl.util.JavaMapping._
+    RouteAdapter { ctx => handler(ctx.request).asScala.map(response => RouteResult.Complete(response.asScala)) }
```

Comment:
Is there a way to delegate to Scala version instead of reimplementing? Or is it too complicated to be worth it?

```suggestion
    RouteAdapter { ctx => handler(ctx.request).asScala.fast.map(response => RouteResult.Complete(response.asScala)) }
```